src/lib/y2security/lsm/config.rb

```diff
@@ -0,0 +1,167 @@
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "yast2/execute"
+require "y2security/lsm/base"
+require "y2security/lsm/app_armor"
+require "y2security/lsm/none"
+require "y2security/lsm/selinux"
+
+Yast.import "Stage"
+Yast.import "ProductFeatures"
+
+module Y2Security
+  module LSM
+    # This class allows to check and select one of the supported Linux Security Modules (LSM)
+    class Config
+      include Yast::Logger
+      include Singleton
+
+      RUNNING_PATH = "/sys/kernel/security/lsm".freeze
+      SUPPORTED = [None, Selinux, AppArmor].freeze
+
+      # @return [None, AppArmor,Selinux, nil] selected module
+      attr_accessor :selected
+      # @return [Boolean] Whether LSM can be configured by the user or not
+      attr_accessor :configurable
+
+      def initialize
+        supported.each { |m| self.class.send(:define_method, m.id) { m } }
+      end
+
+      # Select the LSM to be used based in the one defined in the control file using apparmor as
+      # fallback in case that no one is selected
+      def propose_default
+        log.info("The settings are #{product_feature_settings.inspect}")
+        selected = product_feature_settings.fetch(:select, "apparmor")
+
+        select(selected)
+      end
+
+      # Select the module with the given id when it is selectable.
+      #
+      # @param id [String,Symbol] LSM id to be selected
+      def select(id)
+        @selected = selectable.find { |m| m.id == id.to_sym }
+        log.info @selected ? "Selected LSM with id: #{id}" : "No LSM selected with id: #{id}"
+        @selected
+      end
+
+      # Convenience method to obtain a list of the supported and selectable modules
+      #
+      # @return [Array<Y2Security::LSM::Base] array of supported and selectable LSMs
+      def selectable
+        supported.select(&:selectable?)
+      end
+
+      # Returns the needed patterns for the selected LSM or an empty array if no one is selected
+      #
+      # @return [Array<Sting>]
+      def needed_patterns
+        return [] unless selected
+
+        selected.needed_patterns
+      end
+
+      # Convenience method to save the configuration for the selected LSM
+      def save
+        return false unless selected
+
+        supported.each(&:reset_kernel_params)
+        selected.save
+      end
+
+      # Obtains the supported and active Linux Security Major Module from the running system
+      #
+      # @return [Y2Security::LSM::Base]
+      def from_system
+        active.first
+      end
+
+      # Reads which Linux Security module is active and its configuration if needed but only in a
+      # running system as during the installation there is no active module at all returning false
+      # in that case
+      #
+      # @return [Boolean] whether the configuration was read or not
```

Comment:
```suggestion
      # Reads which Linux Security Module is active and its configuration in a running system
      #
      # @note: during the installation there is **no** active LSM
      #
      # @return [Boolean] false during installation;
      #                   whether the configuration was read or not otherwise.
```